### PR TITLE
[Docs] Improves matmul and mandelbrot parallel notebooks and examples

### PR DESCRIPTION
**Pull request by jackos**
- Uses the new ComplexSIMD and Tensor type from the std lib for mandelbrot instead of implementing them from scratch
- Speeds mandelbrot up by 168x using simplified version from Abdul's blog post, and parallel now 20x faster than sequential
- Matmul notebook now 73,000x faster than python
- Adds a mandelbrot.mojo like the matmul.mojo

### CODE REVIEW

File: examples/matmul.mojo [hunk 1]
```diff
@@ -202,31 +205,33 @@ fn matmul_tiled_unrolled_parallelized(C: Matrix, A: Matrix, B: Matrix):
...
+        fn test_fn():
+            _ = func(C, A, B, rt)
+
+        let secs = Float64(Benchmark().run[test_fn]()) / 1_000_000_000
+        # Prevent the matrices from being freed before the benchmark run
+        _ = (A, B, C)
+        let gflops = ((2 * M * N * K) / secs) / 1e9
+        let speedup: Float64 = gflops / base_gflops
+        # print(gflops, "GFLOP/s", speedup, " speedup")
+        print(str)
```
Comment (dualvtable) [inline examples/matmul.mojo:233]: Non-blocking suggestion, but at some point, we probably want the pretty print fn to reduce the precision. Perhaps in the next update:

```python
def float_printer(f: Float32):
    let builtin = Python.import_module('builtins')
    let pyf = builtin.float(f)
    builtin.print(builtin.str("{:4.2f}").format(pyf))
```

**Review by dualvtable — APPROVED** [review]
LGTM - thanks for the changes!

File: examples/matmul.mojo (hunk 1 above)
Comment (jackos) [reply examples/matmul.mojo:233]: Yes good idea, maybe we can even pop that in the std lib :)

**Comment by lattner** [comment]
Wowwow, nice Jack!

**Pull request merged**